src/etc/inc/openvpn.inc

```diff
@@ -1575,21 +1575,33 @@ function openvpn_resync_all($interface = "") {
 		log_error(gettext("Resyncing OpenVPN instances."));
 	}
 
+	// Check if OpenVPN clients and servers require a resync.
 	if (is_array($config['openvpn']['openvpn-server'])) {
 		foreach ($config['openvpn']['openvpn-server'] as & $settings) {
-			if ($interface <> "" && $interface != $settings['interface']) {
-				continue;
+			$mode_id = "server" . $settings['vpnid'];
+			$fpath = "{$g['varetc_path']}/openvpn/{$mode_id}.interface";
+			$gw_group_change = FALSE;
+			$ip_change = ($interface === $settings['interface']);
+			if (file_exists($fpath)) {
+				$gw_group_change = (trim(file_get_contents($fpath), " \t\n") != get_failover_interface($settings['interface']));
+			}
+			if ($interface === "" || $ip_change || $gw_group_change) {
+				openvpn_resync('server', $settings);
 			}
-			openvpn_resync('server', $settings);
 		}
 	}
-
 	if (is_array($config['openvpn']['openvpn-client'])) {
 		foreach ($config['openvpn']['openvpn-client'] as & $settings) {
-			if ($interface <> "" && $interface != $settings['interface']) {
-				continue;
+			$mode_id = "client" . $settings['vpnid'];
+			$fpath = "{$g['varetc_path']}/openvpn/{$mode_id}.interface";
+			$gw_group_change = FALSE;
+			$ip_change = ($interface === $settings['interface']);
```

Comment:
Same comment for previous definition of `$ip_change`

Reply:
See comment above.